src/app/settings/components/repositoryLocationListItem.tsx

```diff
@@ -51,22 +76,116 @@ export default class RepositoryLocationListItem extends React.Component<Reposito
                     />
                 }
                 {item.repositoryLocationType === REPOSITORY_LOCATION_TYPE.Local &&
-                    <MaskedCopyableTextFieldContainer
-                        ariaLabel={context.t(ResourceKeys.settings.modelDefinitions.repositoryTypes.local.textBoxLabel)}
-                        label={context.t(ResourceKeys.settings.modelDefinitions.repositoryTypes.local.textBoxLabel)}
-                        labelCallout={context.t(ResourceKeys.settings.modelDefinitions.repositoryTypes.local.infoText)}
-                        value={item.value}
-                        allowMask={false}
-                        readOnly={false}
-                        required={true}
-                        onTextChange={this.props.onLocalFolderPathChanged}
-                        placeholder={context.t(ResourceKeys.settings.modelDefinitions.repositoryTypes.local.placeholder)}
-                        setFocus={!item.value}
-                    />
+                   this.renderLocalFolderItem(context)
                 }
             </div>);
     }
 
+    private renderLocalFolderItem = (context: LocalizationContextInterface) => {
+        return(
+                <>
+                    <div className="labelSection">
+                        <LabelWithRichCallout
+                            calloutContent={context.t(ResourceKeys.settings.modelDefinitions.repositoryTypes.local.infoText)}
+                            required={true}
+                        >
+                            {context.t(ResourceKeys.settings.modelDefinitions.repositoryTypes.local.label)}
+                        </LabelWithRichCallout>
+                    </div>
+                    <TextField
+                        className="local-folder-textbox"
+                        label={context.t(ResourceKeys.settings.modelDefinitions.repositoryTypes.local.textBoxLabel)}
+                        ariaLabel={context.t(ResourceKeys.settings.modelDefinitions.repositoryTypes.local.textBoxLabel)}
+                        value={this.state.currentFolder}
+                        readOnly={true}
+                    />
+                    <DefaultButton
+                        text={context.t(ResourceKeys.settings.modelDefinitions.repositoryTypes.local.folderPicker.command.openPicker)}
+                        ariaLabel={context.t(ResourceKeys.settings.modelDefinitions.repositoryTypes.local.folderPicker.command.openPicker)}
+                        onClick={this.onShowFolderPicker}
+                    />
+                    {this.renderFolderPicker(context)}
+                </>
+            );
+    }
+
+    private readonly onShowFolderPicker = () => {
```

Comment:
why are these functions readonly ? these fns are changing state

Reply:
There use to have a linter in this repo that would yell if the functions are not readonly. Didn't pay attention when did this rule get removed, so I was blindly c&p. Let me update it.